sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -158,48 +155,94 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
 
     private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
         CosmosEndToEndOperationLatencyPolicyConfig config = request.requestContext.getEndToEndOperationLatencyPolicyConfig();
+        List<String> preferredRegions = this.transportClient.getGlobalEndpointManager().getPreferredRegions();
+        AvailabilityStrategy strategy = config.getAvailabilityStrategy();
+        if (strategy == null){
+            strategy = new DefaultAvailabilityStrategy();
+        }
+        if (strategy instanceof ThresholdBasedAvailabilityStrategy) {
+                speculativeProcessor = new ThresholdBasedSpeculation();
```

Comment:
There can be race conditions with this check. Should this be atomic operation?

Reply:
Got rid of globals